src/main/java/com/manardenza/dao/AbstractDao.java

```diff
@@ -2,27 +2,78 @@
 
 import com.manardenza.entity.AbstractObject;
 
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
 import java.util.ArrayList;
 import java.util.List;
 
 public abstract class AbstractDao<T extends AbstractObject> {
 
     protected List<T> database = new ArrayList<>();
+    protected File databaseFile;
+    protected FileOutputStream fos = null;
+    protected ObjectOutputStream oos = null;
+    protected FileInputStream fis = null;
+    protected ObjectInputStream ois = null;
+
+    {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Move the contents of this initializer to a standard constructor or to field initializers. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1171)